src/main/java/com/linecorp/armeria/common/ServiceInvocationContext.java

```diff
@@ -43,9 +45,35 @@
 
     /**
      * Returns the context of the invocation that is being handled in the current thread.
+     *
+     * @throws IllegalStateException if the context is unavailable in the current thread
      */
-    public static Optional<ServiceInvocationContext> current() {
-        return Optional.ofNullable(context.get());
+    public static ServiceInvocationContext current() {
+        final ServiceInvocationContext ctx = context.get();
+        if (ctx == null) {
+            throw new IllegalStateException(ServiceInvocationContext.class.getSimpleName() + " unavailable");
+        }
+        return ctx;
+    }
+
+    /**
+     * Maps the context of the invocation that is being handled in the current thread.
+     *
+     * @param mapper the {@link Function} that maps the invocation
+     * @param fallback the {@link Supplier} that provides the value when the context is unavailable
+     *                 in the current thread. If {@code null}, {@code () -> null} is used instead.
```

Comment:
We've been using Optional mostly, but if we're adding nullable parameters, strongly recommend adding jsr305 as an optional dependency and use @Nullable for better documentation and the possibility of null checks.

http://mvnrepository.com/artifact/com.google.code.findbugs/jsr305